### PR TITLE
Feature/add php control structures

### LINKED ISSUE #111: Add PHP control structures
# Description

The control structures (loops, ifs, etc.) in PHP are presently not on Code Thesaurus. I would love to see this added.

# Requirements

* Copy `web/thesauruses/_meta/control_structures.json` to a `web/thesauruses/php/` directory
* Edit the control_structures.json file to accurately reflect PHP and the latest major version of it.

# Additional Notes

Please leave comments on this issue to ask questions or to claim the issue as yours. I'll keep it yours for a week then remove you in case someone else wants to work on it. (You can always reclaim it.)

Check out [CONTRIBUTING.md](CONTRIBUTING.md) for more information on how to contribute. 

Check out [https://github.com/codethesaurus/docs/blob/main/docs/thesaurus/index.md](https://github.com/codethesaurus/docs/blob/main/docs/thesaurus/index.md) for information on more specifics of implementing this.

### PR DESCRIPTION
**Pull request by binarygary**
Closes #111 

Hello!

This adds PHP control structure documentation.
I tested locally but I couldn't figure out how to view PHP control structures. I tried updating web/thesauruses/meta_info.json but it seems like there is some sort of db import step I'm missing due to my lack of django/python knowledge. 

I'm not sure what to do about the foreach loop as in PHP arrays can be associative in which case one could have a different structure. I added a foreach associative example but we may want to take another approach.

Oh, I'm also not sure if I was supposed to modify web/thesauruses/meta_info.json or not...

Thanks,
Gary

### CODE REVIEW

**Comment by geekygirlsarah** [comment]
Cool, I'll try to review this weekend. (I'm behind on a few of them and really need to do this...)

CT is definitely not fully complete on all aspects of every language. I know the meta files need to grow as more types of things can go in (because yeah, I don't think foreach loops are in there, among other types) and I need to get that. Unless you're up for the challenge. (It's not an issue yet but no reason you can't file an issue and take it too. Only if you want.)

**Comment by binarygary** [comment]
I'm happy to riff on php stuff. Let's see where this PR and the other 2 issues I am working on end up. 

Really excited to find this project. I was looking for something like it a couple of years ago.

**Comment by geekygirlsarah** [comment]
> Really excited to find this project. I was looking for something like it a couple of years ago.

I had been looking for something like this for like 5 years. No one really had it (I found some cheat sheets but nothing really interactive and could compare anything). So here we are!

I'll review all the PRs in the next few days. I'm getting back into the flow of working on CT again.

File: web/thesauruses/php/control_strucutres.json [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+      "name": "While loop",
+      "code": "while( $condition ) {\n    // code to be executed\n}"
+    },
+    "do_while_loop": {
+      "name": "Do/While loop",
+      "code": "do {\n    //executable code\n} while ( $condition );"
+    },
+    "for_loops": {
+      "name": "For loop",
+      "code": "for ( $intial_counter, $test_counter, $increment ) {\n    // executable code\n}"
```
Comment (geekygirlsarah) [inline web/thesauruses/php/control_strucutres.json]: The commas here should be semicolons.

**Comment by geekygirlsarah** [comment]
Just a couple of small changes, and I think it's good! I'm glad you added PHP into the meta as well!

If you took out any thing the language can't do, could you add it back in with just an empty section? (The code might break at the moment if this is missing... 😬 I'm working on it.)

File: web/thesauruses/php/control_strucutres.json (hunk 1 above)
Comment (binarygary) [reply web/thesauruses/php/control_strucutres.json]: Good catch, thanks!

**Comment by binarygary** [comment]
@geekygirlsarah I've fixed the For Loop in this PR as well as added back the empty examples I removed.

File: web/thesauruses/php/control_strucutres.json (hunk 1 above)
Comment (geekygirlsarah) [reply web/thesauruses/php/control_strucutres.json]: Easy notice since I find I tend to do this from time to time too! (PHP, C++, JS, whatever)

**Comment by geekygirlsarah** [comment]
This should now be good to merge. Thank you!

**Pull request merged**